[Enhancement] - Automatically download world otbm in folder if file doesn't exist on server start

omeranha opened the pull request:
# Description
  - [X] New feature (non-breaking change which adds functionality)

omeranha committed:
first commit, working

omeranha committed:
sonar, i guess

omeranha committed:
Update map.cpp

gesior commented:
@omeranha 
What? Why? What problem are you trying to fix?
Why do you download otservbr map in canary source?
**Why do you download it from some dropbox, not official otservbr repository** `https://github.com/opentibiabr/otservbr-global/raw/main/data/world/world.zip`?

omeranha commented:
> @omeranha What? Why? What problem are you trying to fix? Why do you download otservbr map in canary source? **Why do you download it from some dropbox, not official otservbr repository** `https://github.com/opentibiabr/otservbr-global/raw/main/data/world/world.zip`?

first: I didn't talk about fixing any problem, didn't you see the Enhancement label in the title?
second:that's first commit, and I'll improve to download both maps, we are discussing this.
third: since you don't know/understand anything about our projects, i will explain it to you, this dropbox link is the new official download of our updated global map, in the next release the we will remove the map zip from repo.

Znote commented:
> third: since you don't know/understand anything about our projects, i will explain it to you, this dropbox link is the new official download of our updated global map, in the next release the we will remove the map zip from repo.

This violates dropbox terms of service. 
This is also a security vulnerability. 
And generally a terrible idea.

dudantas commented on src/map/map.cpp:
```diff
@@ -51,7 +52,19 @@ bool Map::extractMap(const std::string& identifier) const {
+		SPDLOG_INFO("Downloading world.zip to world folder");
+		FILE *fp = fopen("data/world/world.zip", "wb");
+		curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
+		curl_easy_setopt(curl, CURLOPT_URL, "https://www.dropbox.com/s/nmc8w82one8mmp9/world.zip?dl=1");
```
As I suggested in the discord group, the map file is already in the release, feel free to update the link

```suggestion
		curl_easy_setopt(curl, CURLOPT_URL, "https://github.com/opentibiabr/otservbr-global/releases/download/patch-v1.3.1/world.zip");
```

dudantas committed:
Update link to release file

omeranha committed:
hmm

dudantas committed:
Add CURLOPT_SSLVERSION

dudantas commented:
> > third: since you don't know/understand anything about our projects, i will explain it to you, this dropbox link is the new official download of our updated global map, in the next release the we will remove the map zip from repo.
> 
> This violates dropbox terms of service. This is also a security vulnerability. And generally a terrible idea.

I marked it as outdated as we update to stay in the releases. This was something that we had already defined internally as the best alternative, even by creating a "history" linking the map to each release created, but avoiding increasing the history of the repository with binary. Several days ago we had already defined that it would be in the releases, we just hadn't updated it yet. My suggestion is that when you have this type of doubt, look for our discord, because it is the place where we usually discuss these matters, but I believe you know that because you participate there and you must have noticed that it is very busy. Anyway, thanks for your observation, even though this doesn't violate any dropbox terms, it wasn't our idea to keep it there when merging.

omeranha committed:
Merge branch 'main' into download-map

omeranha committed:
Update map.cpp

omeranha committed:
Update map.cpp

dudantas committed:
Merge branch 'main' into download-map

dudantas committed:
Improvement in map download

Added toggleMapDownload and mapDownloadUrl tags on configmanager for help with download map

omeranha committed:
fixing map download
